scrape/main.py

```diff
@@ -0,0 +1,115 @@
+import requests
+import logging
+from bs4 import BeautifulSoup as soup
+from datetime import datetime
+import csv
+import os
+
+def get_cookie(username, password):
+    """Fetches a login cookie from the NYT api
+    Args:
+        username (string): Username (should be an email address).
+        password (string): Password to use.
+
+    Returns: String containing the login cookie fetched using the provided credentials.
+    """
+
+    login_resp = requests.post(
+        'https://myaccount.nytimes.com/svc/ios/v2/login',
+        data={
+            'login': username,
+            'password': password,
+        },
+        headers={
+            'User-Agent': 'Crosswords/20191213190708 CFNetwork/1128.0.1 Darwin/19.6.0',
+            'client_id': 'ios.crosswords',
+        }
+    )
+    login_resp.raise_for_status()
+    for cookie in login_resp.json()['data']['cookies']:
+        if cookie['name'] == 'NYT-S':
+            return cookie['cipheredValue']
+    raise ValueError('NYT-S cookie not found')
+
+
+def scrape_leaderboard(cookie):
+    """Scrapes the leaderboard for the user that the provided cookie corresponds to.
+    Args:
+        cookie (string): Login cookie for the user
+
+    Returns: List all completed entries for the current day.
+    """
+
+    url = 'https://www.nytimes.com/puzzles/leaderboards'
+    response = requests.get(url, cookies={
+        'NYT-S': cookie,
+    })
+    page = soup(response.content, features='html.parser')
+    solvers = page.find_all('div', class_='lbd-score')
+
+    current_datetime = datetime.now()
+    month = str(current_datetime.strftime('%m'))
+    day = str(current_datetime.strftime('%d'))
+    year = str(current_datetime.strftime('%Y'))
+    timestamp = year + '-' + month + '-' + day
+
+    entries = []
+    for solver in solvers:
+        name = solver.find('p', class_='lbd-score__name').text.strip()
+        try:
+            parsed_time = solver.find('p', class_='lbd-score__time').text.strip()
+
+            if parsed_time != '--':
+                minutes, seconds = parsed_time.split(':')
+                time = (60 * int(minutes)) + int(seconds)
+
+                if name.endswith('(you)'):
+                    name = name.split()[0]
```

Comment:
So some of my friends names on the app are just first and last names `Peter Flanagan` can this be 
```py
name.replace('(you)', '').strip()
```

Reply:
Although I guess this is a nonissue because it should only be for me any my username isn't like that